### PR TITLE
chore: reconciling for validUntil option changed

### LINKED ISSUE #1840: In Managed Role, If password not change but password validUntil change should trigger update

### PR DESCRIPTION
**Pull request by litaocdl**
This patch makes sure validuntil option of managed role changing
can reflect to the role in database.

Closes #1840

### CODE REVIEW

File: api/v1/cluster_types.go [hunk 1]
```diff
@@ -1667,9 +1667,9 @@ type RoleConfiguration struct {
 	// +kubebuilder:default:=-1
 	ConnectionLimit int64 `json:"connectionLimit,omitempty"`
 
-	// Date and time after which the role's password is no longer valid.
-	// When omitted, the password will never expire (default).
-	ValidUntil *metav1.Time `json:"validUntil,omitempty"`
+	// Date and time as a timestamp in the RFC3339 standard, after which the role's
+	// password is no longer valid. When omitted, the password will never expire (default).
+	ValidUntil string `json:"validUntil,omitempty"`
```
Comment (jsilvela) [inline api/v1/cluster_types.go:1672]: why change the representation of the timestamp?

File: api/v1/cluster_webhook.go [hunk 2]
```diff
@@ -1858,6 +1858,17 @@ func (r *Cluster) validateManagedRoles() field.ErrorList {
 					role.Name,
 					"This role is reserved for operator use"))
 		}
+
+		if role.PasswordSecret != nil && role.ValidUntil != "" {
+			if _, err := utils.ParseTargetTime(nil, role.ValidUntil); err != nil {
```
Comment (jsilvela) [inline api/v1/cluster_webhook.go:1863]: if we kept the validUntil timestamp as `*metav1.Time`, we'd get parsing / validation for free

**Review by jsilvela — APPROVED** [review]

**Review by sxd — APPROVED** [review]

**Pull request merged**